src/pages/ProductDetails.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import * as api from '../services/api';
+
+export default class ProductDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    const { id: productId } = props;
+
+    this.state = {
+      title: '',
+      price: '',
+      thumbnail: '',
+      attributes: '',
+      loading: true,
+      id: productId,
+    }
+
+    this.detailedProduct = this.detailedProduct.bind(this);
+  }
+
+  async componentDidMount() {
+    const { productId } = this.props.match.params;
+    /* const noSense = api.getCategories();
+    console.log(noSense);
+    const productList = api.getProductsFromCategoryAndQuery(id.productId);attributes
+    console.log(productList); */
+    const product = await api.getProductsFromId(productId);
+    const { title, price, thumbnail, attributes } = product;
+    console.log(product)
+    this.setState({
+      title,
+      price,
+      thumbnail,
+      attributes,
+      loading: false,
+    });
+    console.log(product);
+  }
+
+  detailedProduct() {
+    const { title, price, thumbnail, attributes } = this.state;
+    console.log(this.state);
+    return (
+      <div data-testid="product-detail-link">
+        <div data-testid="product-detail-name">{title}</div>
+        <div>{price}</div>
+        <img src={thumbnail} alt={`image ${title}`} />
+        {attributes.map(({ name, value_name, id }) => {
+          return (<p key={`${id}`}>{`${name}: ${value_name}`}</p>);
+        })}
```

Comment:
aqui seria só uma gestão de indentação

```suggestion
        {
          attributes.map(({ name, value_name, id }) => {
            return (<p key={`${id}`}>{`${name}: ${value_name}`}</p>);
          })
        }
```